code/modules/surgery/surgery.dm

```diff
@@ -553,3 +563,32 @@
 		for(var/reagent in chems_needed)
 			if(target.reagents.has_reagent(reagent))
 				return TRUE
+
+/datum/surgery_step/proc/play_preop_sound(mob/user, mob/living/carbon/target, target_zone, obj/item/tool, datum/surgery/surgery)
+	if(!preop_sound || ismachineperson(target))
+		return
+	var/sound_file_use
+	if(islist(preop_sound))
+		if(istype(tool))
+			for(var/typepath in preop_sound)
+				if(!ispath(typepath) && (typepath in GLOB.surgery_tool_behaviors))
+					sound_file_use = preop_sound[typepath]
+					break
+
+		for(var/typepath in preop_sound)
+			if(ispath(typepath) && istype(tool, typepath))
+				sound_file_use = preop_sound[typepath]
+				break
```

Comment:
What's this second loop for? Could you explain what the logic is supposed to be doing here?

Reply:
In the original PR you wanted me to support global defined tool typepaths while I wanted to keep manual typepaths for ghetto surgery (think using the chainsaw hit sound in place of the surgery saw sound). You can actually still see the conversation we had in the original PR though some of it might have transpired on discord.

To answer your question the second loop is for manual typepaths while the first one is for global defined tool typepaths... I think

Reply:
Oh lmao right! Sorry about that, it's been a minute, and thanks for doing it back on the original PR.
It feels to me like you could probably do this all in one loop, then?

```
for(var/typepath in preop_sound)
	if((ispath(typepath) && istype(tool, typepath)) || (!ispath(typepath) && (typepath in GLOB.surgery_tool_behaviors)))
		sound_file_use = preop_sound[typepath]
		break
```

Reply:
I have tried but I really can't get this to work, even copying your example 1 to 1 doesn't work as manual typepaths break.

All this makes me question my understanding of the original code because it should work with an or operator but it doesn't, the original version still works but this doesn't. 

This is sort of my fault for being lazy and using the code from the original PR from two years ago, I have forgotten why I did it this way and it doesn't help that I am pretty damn rusty at coding(i mean i also sucked back then too lol).

Reply:
oh, hold up. I think I see the problem, we aren't actually checking if the tool has the associated tool behavior. Maybe this will work better?
```
for(var/typepath in preop_sound)
	if((ispath(typepath) && istype(tool, typepath)) || ((typepath in GLOB.surgery_tool_behaviors) && istype(tool) && tool.tool_behaviour == typepath))
		sound_file_use = preop_sound[typepath]
		break
```

Reply:
Yep, that was it. The change is in the branch now!